packages/block-editor/src/store/reducer.js

```diff
@@ -656,6 +679,54 @@ export const blocks = flow(
 
 		return state;
 	},
+
+	// This is the opposite of the order property.
+	//  It's duplicated data used for performance reasons.
+	parents( state = {}, action ) {
+		switch ( action.type ) {
+			case 'RESET_BLOCKS':
+				return mapBlockParents( action.blocks );
+
+			case 'RECEIVE_BLOCKS':
+				return {
+					...state,
+					...mapBlockParents( action.blocks ),
+				};
+
+			case 'INSERT_BLOCKS':
+				return {
+					...state,
+					...mapBlockParents( action.blocks, action.rootClientId || '' ),
+				};
+
+			case 'MOVE_BLOCK_TO_POSITION': {
+				return {
+					...state,
+					[ action.clientId ]: action.toRootClientId || '',
+				};
+			}
+
+			case 'REPLACE_BLOCKS':
+				return {
+					// This should include the parents as well to avoid memory leak
```

Comment:
Similarly, we could be more specific to guide future maintainers on what the memory leak is:

```suggestion
					// This should include the parents to avoid a memory leak
					// caused by orphaned references to blocks.
```

Reply:
This behavior seems particularly important to unit test.

Reply:
I think I want to solve this before merge